### ISSUE TITLE
Training by myself ,the loss=nan

### ISSUE THREAD

**Issue by litingsjj** [sfzhang15/RefineDet#119]
Hi, I use your code to train psacal_voc by myself. But the loss=nan, I use one gpu,and the batch_size=4

**Comment by litingsjj** [comment]
Sorry, find some closed issues that have the same problem, I make batch_size=8, it looks fine, but why?

**Comment by litingsjj** [comment]
@sfzhang15 Sorry, can you tell me the final loss when you finish training?

**Comment by sfzhang15** [comment]
@litingsjj 
about 3

**Comment by litingsjj** [comment]
@sfzhang15  Thanks！

**Comment by litingsjj** [comment]
@sfzhang15  Thanks closed issues! I find same question in closed issues. About mobilenetV2, I increase bn layer, it seems fine, when begin training(loss=25. ). Now I have other questions.
1. I train VGG( you trained) that loss is equal 5~6 about 26K iteration( when 10K iteration ,the loss=5~6). It seems final loss=5~6. I use VGG16_VOC2007_512.py
2. Also loss about mobilenetV1, the loss=10~11, and I think the mobilenetV2 will have same question. Do you know why that happened? Have you some tricks to handle this?
 Appreciate!

**Comment by jasminezz** [comment]
I have the same problem~
I train VGG loss is equal 2~3. 
The loss of SqueezeNet is equal 10~11 , and the prediction is awful.
Wating for your answer @sfzhang15

**Comment by litingsjj** [comment]
@jasminezz How many iterations you trained for VGG?

**Comment by sfzhang15** [comment]
@litingsjj @jasminezz 
1. The final loss (120k iterations) is about 3 for VGG16_VOC07_320.
2. We have not trained RefineDet with MobileNet or SqueezeNet, but we have trained the face detector based on SSD+FPN with MobileNet or SqueezeNet and achieved promising results. Our experience is to add the normalization layer at corresponding pretrained layers with suitable scale values, making the start loss value is normal.

**Comment by litingsjj** [comment]
@sfzhang15 Thanks! I trained it 19K iterations that the loss is equal 6~7. I'm expecting it!

**Comment by jasminezz** [comment]
@litingsjj 120k iterations trained VGG， loss is equal _2~3_
Now, the squeezeNet, I trained 4k iterations that the loss is equal 8.

**Comment by litingsjj** [comment]
@jasminezz 30K iterations trained for mobilenetV2, loss is still equal 6~7. Maybe I should decrease lr 0.001 to 0.0001. In solver.prototxt, stepvalue=80K ,lr=0.0001

**Comment by jasminezz** [comment]
@litingsjj My training is still _7~8_.
Moreover, The Flops of this method is too large. Even if using the Lightest models, Flops is almost 1G. 
I will quit this method, the design cannot used by mobile devices.

**Comment by litingsjj** [comment]
@sfzhang15 Hi, I finish 120K iteration and the loss is equal 5.2, but the result is not so well, actually it's a little bad when I test several images.

**Comment by sfzhang15** [comment]
@litingsjj 
Sorry, since we do not train RefineDet with mobilenetV2, so we can not give you some useful suggestions. Accroding to our experience of training the face detector based on SSD+FPN with MobileNet, adding the normalization layer at corresponding pretrained layers with suitable scale values to make the start loss value normal is one of the keys. You can try to add some normalization layers to make the starting loss same with VGG16.
